docs/developer/v1alpha1-compared-to-v1alpha2.md

```diff
@@ -0,0 +1,65 @@
+# Cluster API v1alpha1 compared to v1alpha2
+
+## Providers
+
+### v1alpha1
+
+Providers in v1alpha1 are generic cloud providers including, aws, gcp, openstack, azure, vsphere and others. The provider 
+vendors in Cluster API's controllers, registers their actuators with the controllers and runs a single manager 
+(with the Cluster API controllers) to complete a Cluster API management cluster.
```

Comment:
I'd probably call out that there is only a singular type of provider in this section and that provider is responsible for both infrastructure management and kubernetes bootstrapping.

Reply:
It might also be good to point out here that provider-specific configuration is embedded within the Cluster and Machine types as `Spec.ProviderSpec` and provider-specific status is embedded as `Status.ProviderStatus`

Reply:
The latter suggestion to specify field types seems like too much detail. The aim for this doc should be an overview of the differences.